src/promo/promo.repository.ts

```diff
@@ -0,0 +1,68 @@
+import { plainToInstance } from 'class-transformer';
+
+import { appDataSource } from '../../data-source';
+import { PromocodeEntity } from './entity/promo.entity';
+import { IAddPromoCode, IPageOptions, IPromoCode, IPromoCodeName, IRemovePromoCode } from './promo.interface';
+import { PromocodeState } from './promocode.enum';
+
+export class PromocodeRepository {
+  async getListOfPromocodes(pageOptions: IPageOptions): Promise<PromocodeEntity[]> {
+    let query = 'SELECT * FROM promocode';
+
+    if (!pageOptions.page) {
+      query = query.concat(' LIMIT ALL;');
+    } else {
+      const skip: number = (pageOptions.page - 1) * pageOptions.limit;
+      query = query.concat(` LIMIT ${pageOptions.limit} OFFSET ${skip}`);
+    }
+    const promocodes: PromocodeEntity[] = await appDataSource.query(query);
+    return plainToInstance(PromocodeEntity, promocodes);
+  }
+
+  async addPromoCode(promocode: IAddPromoCode): Promise<IPromoCode> {
+    const newPromocode: PromocodeEntity[] = await appDataSource.query(
+        `INSERT INTO promocode (name, percent, "isOneTime", "startDate", "endDate", "currentState")
+          VALUES ($1, $2, $3, $4, $5, $6) RETURNING *`,
+        [
+          promocode.name.toLowerCase(),
+          promocode.percent,
+          promocode.isOneTime,
+          promocode.isOneTime ? new Date(promocode.startDate) : null,
+          promocode.isOneTime ? new Date(promocode.endDate) : null,
+          PromocodeState.ACTIVE,
+        ],
+    );
+    return newPromocode[0];
+  }
+
+  async removePromoCode(promocode: IRemovePromoCode): Promise<Boolean> {
+    const response = await appDataSource.query(
+        `UPDATE promocode
+        SET
+          "currentState" = '${ PromocodeState.REMOVED }', 
```

Comment:
don't think we need this status. or do you plan to add ned status for promocodes in future?